catalogue/catalogue-service.yaml

```diff
@@ -0,0 +1,807 @@
+          schema:
+            type: string
+          example: rails
+          description: 'Find Products with matching name (supports wild-card, partial-word match search)'
+        - in: query
+          name: 'filter[in_stock]'
+          schema:
+            type: boolean
+          example: true
+          description: Returns only in_stock products
+        - in: query
+          name: sort_by
+          schema:
+            type: string
+          example: '-updated_at,price'
+          description: |-
+            Sort products based on: <ul>
+              <li>available_on (ascending)</li>
+              <li>delivery_included_price (ascending/descending)</li>
+              <li>price (ascending/descending)</li>
+              <li>updated_at (ascending/descending)</li>
+            </ul> Use <q>-</q> sign to set descending sort, eg. <q>-updated_at</q>
+        - $ref: '#/components/parameters/PageParam'
+        - $ref: '#/components/parameters/PerPageParam'
+      responses:
+        '200':
+          description: Returns a list of Products
+          content:
+            application/vnd.api+json:
+              schema:
+                type: object
+                properties:
+                  products:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Product'
+                  meta:
+                    $ref: '#/components/schemas/ProductListMeta'
+                  list_links:
+                    $ref: '#/components/schemas/ListLinks'
+        '429':
+          $ref: '#/components/responses/429TooManyRequests'
+      summary: List of Products
+      
+  '/catalog/products/{id}':
+    get:
+      description: |-
+        Returns Product details. You can use product ID:
+
+        ```
+        GET /catalog/products/21
+        ```
+      tags:
+        - Products
+      operationId: Show Product
+      parameters:
+        - $ref: '#/components/parameters/IdOrPermalink'
+      responses:
+        '200':
+          description: Returns the requested Product
+          content:
+            application/vnd.api+json:
+              schema:
+                type: object
+                properties:
+                  product:
+                    $ref: '#/components/schemas/Product'
+                  meta:
+                    properties:
+                      breadcrumbs:
+                        $ref: '#/components/schemas/BreadcrumbsMeta'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+        '429':
+          $ref: '#/components/responses/429TooManyRequests'
+      summary: Get Product Details
+
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      description: |-
+        User token to authorize Cart and Checkout requests.
+
+        You can obtain it from [/spree_oauth/token](../../oauth/index.yml) endpoint.
+        It is required to associate Cart with the User.
+
+  schemas:
+    Error:
```

Comment:
Should this make use of the common `Errors` type from https://github.com/Crown-Commercial-Service/ccs-scale-api-definitions/blob/SCA-1516-Catalog-Service-Iteration-1/common/CCS_Standard_API_Schema.yaml#L109 ?

Reply:
Thanks. In a perfect world yes. Initially I just copied the Spree ones (both error object and property names). Agree that the format should align with previously - even if Spree is different I'm sure you are able to translate.

Reply:
Have updated to use the central Error definition (and some others, AgreementID, LotID, Timestamp, Email etc). Have also corrected camelCase properties